src/utils/netutils.cpp

```diff
@@ -811,3 +965,69 @@ bool get_local_ip_address(IpAddress& client_address, int fd) noexcept
 
     return true;
 }
+
+bool find_probe_client(std::string_view probe_client_addresses,
+                       zstring_view source_ip,
+                       bool is_ipv6)
+{
+    if (probe_client_addresses.empty())
+    {
+        return false;
+    }
+
+    sockaddr_storage source_ss;
+
+    if (!get_in_addr_from_ip(source_ss, source_ip.c_str(), is_ipv6))
+    {
+        return false;
+    }
+
+    char probe_ip_buf[INET6_ADDRSTRLEN] = {};
+    auto try_fill_buffer_and_compare_ip = [&](auto start, auto end)
+    {
+        /* check substring size is greater than 0 and less
+           than IPv6 max size (i.e a valid IP in theory) */
+        if (end <= start || end - start >= sizeof(probe_ip_buf))
+        {
+            return false;
+        }
+
+        auto i = 0;
+
+        while (start < end)
+        {
+            probe_ip_buf[i++] = probe_client_addresses[start++];
+        }
+
+        probe_ip_buf[i] = '\0';
```

Comment:
```cpp
static_string<INET6_ADDRSTRLEN> probe_ip_buf;
// ....
return probe_ip_buf.try_assign(addr)
    && compare_binary_ip(source_ss, probe_ip_buf.c_str(), is_ipv6))

/// \return \c false when the input string is greater than \c max_capacity(), otherwise \c true.
bool static_string::try_assign(chars_view str) noexcept
{
    if (str.size() <= N) {
        *this = bounded_array_view<0, N>::assumed(str);
        return true;
    }
    return false;
}
```